upi/vsphere/lb/haproxy.tmpl

```diff
@@ -32,6 +32,10 @@ backend api-server
     %{ for addr in api ~}
     server ${addr} ${addr}:6443 check
     %{ endfor ~}
+    mode https
+    %{ for addr in api ~}
+    server ${addr} ${addr}:6443/readyz check inter 10s fall 3 rise 2
+    %{ endfor ~}
```

Comment:
So for api we need
- mode tcp because we are trying to do L4 balancing
- but have https checks on `/readyz`

seeing https://serverfault.com/questions/924477/haproxy-health-check-for-https-backend
we need something like,
```
  mode tcp
  balance roundrobin
  option httpchk HEAD /readyz
  http-check expect status 200
  option ssl-hello-chk
  server one <1 IP>:6443 check check-ssl verify none
  server two <2 IP>:6443 check check-ssl verify none
```

??